### DISCUSSION TITLE
Position absolute with .prose causes empty space depending on where the element is located

### DISCUSSION THREAD

**Discussion by ivanBell199** [tailwindlabs/tailwindcss#10115] [Help] [answered]
I wanted to make a simple card using TailwindCSS and `.prose`. I found what seems to me a bug. I assign my card `position: relative` and then add an element with `position: absolute`. If in my HTML I put this element in the top, I get undesired empty space in my card.
```
   <article class="prose w-full ring m-4 p-6 relative">
      <span class="absolute -top-4 right-4 px-4 py-2 bg-green-300">This is not OK</span>
      <h2>Heading</h2>
    </article>
```
If I put the element after any other card child, then I get the needed result.

You can check my codepen: [https://codepen.io/iBeliakov/pen/OJwLYWL](url)

I am using CDN `https://cdn.tailwindcss.com?plugins=forms,typography,aspect-ratio,line-clamp`

**Comment by wongjn** [comment, marked as answer]
The white space is from the `margin-top` applied to the `<h2>` element. If the `<h2>` element was the first element of the `prose`, it would not have this margin top, but since there is an element preceding the `<h2>` element, it has margin top, since `prose` assumes there isn't any wildly positioned elements. I don't believe that prose could possibly cover all cases where this occurs, but you can adjust the margins for your particular use case, here are some possible solutions:

## Fix 1

```diff
 <span class="absolute -top-4 right-4 px-4 py-2 bg-green-300">This is not OK</span>
-<h2>Heading</h2>
+<h2 class="mt-0">Heading</h2>
 <p>Content.</p>
```

## Fix 2

```diff
-<article class="w-full ring prose m-4 p-6 relative">
+<article class="w-full ring prose m-4 p-6 relative [&:nth-child(2)]:prose-h2:mt-0">
   <span class="absolute -top-4 right-4 px-4 py-2 bg-green-300">This is not OK</span>
```

## Fix 3

```diff
-<article class="w-full ring prose m-4 p-6 relative">
+<article class="w-full ring prose m-4 p-6 relative [&_.absolute:first-child+*]:mt-0">
   <span class="absolute -top-4 right-4 px-4 py-2 bg-green-300">This is not OK</span>
```

Here's a [Tailwind Play of these solutions as live examples](https://play.tailwindcss.com/Nceet2xSdp).
